apps/pokedex/pokedex.star

```diff
@@ -0,0 +1,54 @@
+"""
+Applet: Pokedex
+Summary: Display a random Pokemon
+Description: Display a random Pokemon along with its height and weight.
+Author: Mack Ward
+"""
+
+load("http.star", "http")
+load("render.star", "render")
+load("schema.star", "schema")
+load("time.star", "time")
+
+NUM_POKEMON = 386
+POKEAPI_URL = "https://pokeapi.co/api/v2/pokemon/{}"
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [],
+    )
+
+def main():
+    id_ = int(NUM_POKEMON * random()) + 1
+    pokemon_url = POKEAPI_URL.format(id_)
+    pokemon = http.get(pokemon_url).json()
```

Comment:
According to their API, we should cache this data: https://pokeapi.co/docs/v2. Seems like we could cache for a few days on a given pokemon. We also should check the response for an error. Happy to land this in the short term and will follow up before launch.

Reply:
@betterengineering, just FYI, I played around with caching for a bit but decided to skip it to avoid the extra complexity. Ideally, both the original request and the follow-up request (for the sprite) should be cached with the same TTL - they should use the same cache key - but I couldn't figure out how to cache a JSON dictionary containing all of the data. Serializing is easy, but I couldn't find a method for JSON deserialization.

Reply:
You can create a new dict with both (base64-encode the sprite) and then use json.encode to encode it into a string that you can put it in the cache.  Then json.decode the cached value and base64-decode the sprite from it.